funcx_sdk/funcx/sdk/client.py

```diff
@@ -124,10 +130,12 @@ def __init__(
         if results_ws_uri is None:
             results_ws_uri = get_web_socket_url(environment)
 
-        self.func_table = {}
+        self.func_table: t.Dict[str, t.Dict] = {}
         self.use_offprocess_checker = use_offprocess_checker
         self.funcx_home = os.path.expanduser(funcx_home)
-        self.session_task_group_id = str(uuid.uuid4())
+        self.session_task_group_id = (
+            task_group_id and str(task_group_id) or str(uuid.uuid4())
+        )
```

Comment:
```suggestion
        self.session_task_group_id = (
            str(task_group_id) if task_group_id is not None else str(uuid.uuid4())
        )
```

Rather than clever use of `and`, why not an inlined if-else?